main.go

```diff
@@ -21,32 +24,120 @@ type vaultExecConfig struct {
 	Path    string `json:"path"`
 }
 
-// Do a case insensitive search for MESOS_TASK_ID
-func getTaskId(envVars []string) (string, error) {
-	for _, envVar := range envVars {
-		res := strings.SplitN(envVar, "=", 2)
-		if strings.ToUpper(res[0]) == "MESOS_TASK_ID" {
-			return res[1], nil
-		}
+func determineScheduler() (string, error) {
+	if os.Getenv("MESOS_TASK_ID") != "" {
```

Comment:
Wait, do you mean to return the "mesos" default when the environment variable is not empty?

Reply:
Oh, maybe I need to a do a better job understanding what this does. Is the idea here saying that you need either one of these set, and then that determines what scheduler it looks for?

Reply:
Yeah. I am just trying to determine by the environment variables set which container scheduler is being used. To use vault-gatekeeper the MESOS_TASK_ID var must be set and Mesos always sets it. The other env is just something that k8s sets that I can check if the container is running in k8s.